packages/sync-actions/test/product-types-sync-attribute-hints.spec.js

```diff
@@ -0,0 +1,492 @@
+                action: 'changeLabel',
+                attributeName: 'attribute-name',
+                label: { en: 'next-attribute-label' },
+              },
+            ],
+            [
+              'inputTip',
+              {
+                action: 'setInputTip',
+                attributeName: 'attribute-name',
+                inputTip: {
+                  en: 'next-input-tip',
+                },
+              },
+            ],
+            [
+              'inputHint',
+              {
+                action: 'changeInputHint',
+                attributeName: 'attribute-name',
+                newValue: 'MultiLine',
+              },
+            ],
+            [
+              'isSearchable',
+              {
+                action: 'changeIsSearchable',
+                attributeName: 'attribute-name',
+                isSearchable: true,
+              },
+            ],
+            [
+              'attributeConstraint',
+              {
+                action: 'changeAttributeConstraint',
+                attributeName: 'attribute-name',
+                newValue: 'None',
+              },
+            ],
+          ]
+          it.each(changes)(
+            'should generate update action for %s',
+            (name, expectedUpdateAction) => {
+              expect(updateActions).toEqual(
+                expect.arrayContaining([expectedUpdateAction])
+              )
+            }
+          )
+        })
+      })
+      describe('without next', () => {
+        beforeEach(() => {
+          attributeDefinitionDraftItem = createAttributeDefinitionDraftItem({
+            next: undefined,
+          })
+          updateActions = sync.buildActions(
+            {},
+            {},
+            {
+              nestedValuesChanges: {
+                attributeDefinitions: [attributeDefinitionDraftItem],
+              },
+            }
+          )
+        })
+        it('should match snapshot', () => {
+          expect(updateActions).toMatchSnapshot()
+        })
+        it('should generate `removeAttributeDefinition` update-action', () => {
+          expect(updateActions).toEqual(
+            expect.arrayContaining([
+              {
+                action: 'removeAttributeDefinition',
+                name: 'attribute-name',
+              },
+            ])
+          )
+        })
+      })
+    })
+    describe('without previous', () => {
+      beforeEach(() => {
+        attributeDefinitionDraftItem = createAttributeDefinitionDraftItem({
+          previous: undefined,
+        })
+        updateActions = sync.buildActions(
+          {},
+          {},
+          {
+            nestedValuesChanges: {
+              attributeDefinitions: [attributeDefinitionDraftItem],
+            },
+          }
+        )
+      })
+      it('should match snapshot', () => {
+        expect(updateActions).toMatchSnapshot()
+      })
+      it('should generate `addAttributeDefinition` update-action', () => {
```

Comment:
no they are not equal. A snapshot can break (because another action), where as what matters here is  that `addAttributeDefinition` always is present.

Also when a snapshot break, it's not uncommon that devs just updates the snap and move on with potentially letting a break slide

Reply:
If you just update the snapshot without investigating it then it is completely pointless and this test does not have to be there at all. 

And if it does not generate the `addAttributeDefinition` than it will break also snapshot so that's why I thought that this test is not needed.